### DISCUSSION TITLE
(Auto) Cleaning of temporary rules

### DISCUSSION THREAD

**Discussion by BetweenTheKeyboardAndChair** [evilsocket/opensnitch#765] [Ideas]
Issue: I'm finding that the temporary rules section becomes enormous over time.  Cleaning out the old rules can take 5m+, since one can only delete a page worth of rules at a time.

Possibilities:

1) Enable the multi-row selection mechanism to cross screens.  At the moment it only selects within a page.  As a concrete example, suppose I have 1000 items I want to delete, and I can see 50 on the screen at a time.  If I select #1, and then scroll down and select #1000, I actually end up with only 950-1000 selected. (top to bottom of the visible screen).

2) A timed cleaning similar to events.  e.g. Delete all temporary rules that have not been active for at least N days.
This would be my preference.  Any temporary rules shorter than "until reboot" can go after a day or two.

Thank you for all the work.  The app is great.

**Comment by BetweenTheKeyboardAndChair** [comment]
Rewriting without the hash marks, so git doesn't "help".  It won't let me edit the text with errors above.

1. Enable the multi-row selection mechanism to cross screens. At the moment it only selects within a page. As a concrete example, suppose I have 1000 items I want to delete, and I can see 50 on the screen at a time. If I select row 1, and then scroll down and select  row 1000, I actually end up with only 950-1000 selected. (top to bottom of the visible screen).

**Comment by gustavo-iniguez-goya** [comment]
hey @BetweenTheKeyboardAndChair , sorry for the delayedd response. 

Would this option help? Preferences -> UI

![image](https://user-images.githubusercontent.com/2742953/202020835-e8c22d6c-06d8-493c-b1a3-3b215b3b5b2b.png)

**Comment by BetweenTheKeyboardAndChair** [comment]
Re: "Don't save rules of duration"

I tried that option before posting my request.  It hid too much.
1. I want to know what is currently allowed/blocked.
2. I want an easy way to fix a misclick - i.e. I blocked (my default) the 
wrong thing.
3. I use "allow/deny until reboot" as a testing method.  I convert most of 
them to permanent before I reboot.

The current UI allows for #2, and #3.  #1 is hard.  There is no sort order 
that makes it easy to see
"active".  A sort by time can hide the rule I set with a 60m expiration 30m 
ago; there are often too many
5m rules that have come/gone since then.  When I'm looking at the active, 
rules I'm almost always
interested in the temporary active rules.  I would not want them 
interleaved/overwhelmed by the permanent rules.

Thanks for working on the tool.
